### PR TITLE
Added test case for bloom filter

### PR DESCRIPTION
**Pull request by martinpinto**
Added test case for bloom filter. WIP more to come

### CODE REVIEW

File: src/sketches/bloom_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"datamodel"
+	"utils"
+)
+
+func TestAddBloom(t *testing.T) {
+	utils.SetupTests()
+	defer utils.TearDownTests()
+
+	info := datamodel.NewEmptyInfo()
+	info.Properties.Capacity = 1000000000
```
Comment (seiflotfy) [inline src/sketches/bloom_test.go]: why would you allocate over 1.12GB in a unit test?

File: src/sketches/bloom_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		[]byte("cyclops"),
+		[]byte("cyclops"),
+		[]byte("cyclops"),
+		[]byte("havoc")}
+
+	if _, err := sketch.Add(values); err != nil {
+		t.Error("expected no errors, got", err)
+	}
+
+	if res, err := sketch.Get([][]byte{[]byte("cyclops")}); err != nil {
```
Comment (seiflotfy) [inline src/sketches/bloom_test.go]: you are only testing for positives.
do the following instead: create an "expected" map[string]bool with keys and values of what you expect for each key, then just do a Get on all keys of the map. And check each member in a for loop if the Member.Name == expected[Member.Name]

File: src/sketches/bloom_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		t.Error("expected no errors, got", err)
+	} else if res.([]*datamodel.Member)[0].Member != true {
+		t.Error("expected 'cyclops' count == true, got", res.([]*datamodel.Member)[0].Member)
```
Comment (seiflotfy) [inline src/sketches/bloom_test.go]: change this please from count == true to member == true, also make sure with the above comment you need to adapt the if condition as well as the error if it goes wrong

File: src/sketches/bloom_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+	values := make([][]byte, 10)
+	for i := 0; i < 1024; i++ {
+		avenger := "avenger" + strconv.Itoa(i)
+		values = append(values, []byte(avenger))
+	}
+
+	for i := 0; i < 1024; i++ {
+		info := datamodel.NewEmptyInfo()
+		info.Properties.Capacity = 1000000000
```
Comment (seiflotfy) [inline src/sketches/bloom_test.go]: please no 1 GB

**Comment by seiflotfy** [comment]
The tests are not running

File: src/sketches/bloom_test.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+		t.Error("expected no errors, got", err)
+	}
+
+	expected := [][]byte{
+		[]byte("sabertooth"),
+		[]byte("thunderbolt"),
+		[]byte("havoc"),
+		[]byte("cyclops")}
+
+	for i := 0; i < len(expected); i++ {
```
Comment (seiflotfy) [inline src/sketches/bloom_test.go]: You can do a Get on all the values so the for loop is in the wrong place here.

File: src/sketches/bloom_test.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+		[]byte("cyclops"),
+		[]byte("cyclops"),
+		[]byte("cyclops"),
+		[]byte("havoc")}
+
+	if _, err := sketch.Add(values); err != nil {
+		t.Error("expected no errors, got", err)
+	}
+
+	expected := [][]byte{
```
Comment (seiflotfy) [inline src/sketches/bloom_test.go]: This is not write, its just a repeat of the values. I asked for a map[string]bool
so basically {"sabertooth": true, ""thunderbolt", true, ....} there is no harm in adding something that expects a false.

**Comment by seiflotfy** [comment]
Will continue reviewing once this is taken care of.

**Comment by martinpinto** [comment]
#123 pushing to master instead of revamp

**Pull request closed without merging**